File: arraypairsum.py
# Time Complexity: O(n)
# Space Complexity: O(1)
# Approach: Sort the numbers. The min in each pair will be in alternate positions. So take its sum and that will be the result
class Solution:
    def arrayPairSum(self, nums: List[int]) -> int:
        nums.sort()
        result = 0
        for i in range(0,len(nums),2):
            result += nums[i]
            
        return result

File: arrayPairSum.java
// Time Complexity: O(nlog(n))
// Space Complexity: O(1)
// Approach: Idea here is ot sort the elements so that bigger elements are prevented to be matched
// with smaller elements and there by taking small values from small pair and big values from big pairs
class Solution {
    public int arrayPairSum(int[] nums) {
        Arrays.sort(nums);
        int sum = nums[0];
        for(int i=2; i< nums.length-1; i=i+2) {
            sum+=nums[i];
        }
        return sum;
    }
}